db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java

```diff
@@ -526,6 +526,32 @@ public void bindExpressions(FromList fromListParam) throws StandardException{
             return;
         }
 
+        // Evaluate expressions with constant operands here to simplify the
+        // query tree and to reduce the runtime cost. Do it before optimize()
+        // since the simpler tree may have more accurate information for
+        // the optimizer. (Example: The selectivity for 1=1 is estimated to
+        // 0.1, whereas the actual selectivity is 1.0. In this step, 1=1 will
+        // be rewritten to TRUE, which is known by the optimizer to have
+        // selectivity 1.0.)
+        // This is also done before predicate simplification to enable
+        // more predicates to be pruned away.
+        accept(new ConstantExpressionVisitor());
```

Comment:
Should we skip the nested SelectNode here? Given that SelectNode.bindExpressions() is recursively called, it seems that we could visit a nested SelectNode with the ConstantExpressionVisitor multiple times.

Reply:
@yxia92 I agree.  Made the change.